components/interop/interop-service/src/main/kotlin/net/corda/interop/service/impl/FacadeToFlowMapperServiceImpl.kt

```
@@ -0,0 +1,73 @@
+package net.corda.interop.service.impl
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import net.corda.cpiinfo.read.CpiInfoReadService
+import net.corda.interop.service.InteropFacadeToFlowMapperService
+import net.corda.virtualnode.HoldingIdentity
+import net.corda.virtualnode.read.VirtualNodeInfoReadService
+import org.osgi.service.component.annotations.Activate
+import org.osgi.service.component.annotations.Component
+import org.osgi.service.component.annotations.Reference
+import org.slf4j.LoggerFactory
+
+
+@Component(service = [InteropFacadeToFlowMapperService::class])
+class FacadeToFlowMapperServiceImpl @Activate constructor(
+    @Reference(service = VirtualNodeInfoReadService::class)
+    private val virtualNodeInfoReadService: VirtualNodeInfoReadService,
+    @Reference(service = CpiInfoReadService::class)
+    private val cpiInfoReadService: CpiInfoReadService
+) : InteropFacadeToFlowMapperService {
+    companion object {
+        private const val FACADE_TO_FLOW_MAPPING = "FACADE_TO_FLOW_MAPPING"
+        private val logger = LoggerFactory.getLogger(this::class.java.enclosingClass)
+        private val mapper = jacksonObjectMapper()
+    }
+
+    override fun getFlowName(
+        destinationIdentity: HoldingIdentity,
+        facadeId: String,
+        facadeName: String
+    ): String? {
+        logger.info(
+            "Fetching flow name for holding identity : ${destinationIdentity.x500Name} " +
+                    "facade id : $facadeId facade name : $facadeName"
+        )
+        // Now we are using hardcoded facade definition and
+        // the assumption is we will get the required facade to flow mapping configuration as a part of CPIMetaData.
+        // The following code snippet suggests how we can read it and when it will be available.
+        val vNodeInfo = virtualNodeInfoReadService.get(destinationIdentity)
+        val content = if (vNodeInfo != null) {
+            checkNotNull(vNodeInfo) {
```

This is a bit odd as you're doing 2 null checks on the same object.

---

The first if is just to use the fallback on hardcoded mapping. The second one I kept deliberately so that when we will remove the hardcoded one, we need to throw the exception if vNode is null. Second one is not getting called now.